### PR TITLE
[Prow CI workflow Enablement] Use MCE create HostedCluster on BM Disconnected + agent

### PR DESCRIPTION
**Pull request by LiangquanLi930**
This workflow helps us to achieve:
1. Using IPV6 single stack; [OCPSTRAT-373](https://issues.redhat.com/browse/OCPSTRAT-373)
2. Install on disconnected Env. [HOSTEDCP-477](https://issues.redhat.com/browse/HOSTEDCP-477)

Based on this, I will work on Dual stack + Disconnected.
Job link: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/44946/rehearse-44946-periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-conformance/1718844306439016448

### CODE REVIEW

**Comment by LiangquanLi930** [comment]
[OCPQE-16668](https://issues.redhat.com/browse/OCPQE-16668)

**Comment by LiangquanLi930** [comment]
[OCPQE-16671](https://issues.redhat.com/browse/OCPQE-16671)

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-conformance

File: ci-operator/step-registry/hypershift/agent/create/config-dns/hypershift-agent-create-config-dns-commands.sh [hunk 1]
```diff
@@ -12,20 +12,41 @@ CLUSTER_NAME="$(echo -n $PROW_JOB_ID|sha256sum|cut -c-20)"
...
 set -x
 EOF
+elif [ "$IP_STACK" = "v6" ]; then
+  # shellcheck disable=SC2087
+  ssh "${SSHOPTS[@]}" "root@${IP}" bash - << EOF |& sed -e 's/.*auths.*/*** PULL_SECRET ***/g'
+WORKER_IP=\$(oc get node -lnode-role.kubernetes.io/worker="" -o jsonpath='{.items[0].status.addresses[?(@.type=="InternalIP")].address}')
+BASEDOMAIN=\$(oc get dns/cluster -ojsonpath="{.spec.baseDomain}")
+HOSTEDCLUSTER_NAME=\$(cat /home/hostedcluster_name)
+echo "address=/api.\$HOSTEDCLUSTER_NAME.\$BASEDOMAIN/\$WORKER_IP" >> /etc/NetworkManager/dnsmasq.d/openshift-ostest.conf
+echo "address=/api-int.\$HOSTEDCLUSTER_NAME.\$BASEDOMAIN/\$WORKER_IP" >> /etc/NetworkManager/dnsmasq.d/openshift-ostest.conf
```
Comment (jparrill) [inline ci-operator/step-registry/hypershift/agent/create/config-dns/hypershift-agent-create-config-dns-commands.sh:39]: The WORKER_IP comes from the MGMT Cluster's worker or HostedCluster worker? (it should point to the MGMT Nodes where the HCP API is running)

Is this working fine?

File: ci-operator/step-registry/hypershift/mce/agent/disconnected/nfs/hypershift-mce-agent-disconnected-nfs-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  from: assisted-service-scripts
+  commands: hypershift-mce-agent-disconnected-nfs-commands.sh
+  resources:
+    requests:
+      cpu: 1000m
+      memory: 500Mi
+  timeout: 30m0s
+  grace_period: 5m0s
+  documentation: |-
+    Deploy nfs( network file system) provisioner for HostedCluster ControlPlane's etcd
```
Comment (jparrill) [inline ci-operator/step-registry/hypershift/mce/agent/disconnected/nfs/hypershift-mce-agent-disconnected-nfs-ref.yaml:12]: small nit: `s/nfs( network file system)/nfs (network file system)/`

File: ci-operator/step-registry/hypershift/agent/create/config-dns/hypershift-agent-create-config-dns-commands.sh (hunk 1 above)
Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/agent/create/config-dns/hypershift-agent-create-config-dns-commands.sh:39]: yep, WORKER_IP  comes from the MGMT Cluster's worker

**Comment by LiangquanLi930** [comment]
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/44946/rehearse-44946-periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-agent-ovn-mce-disconnected-conformance/1718844306439016448

File: ci-operator/step-registry/hypershift/azure/destroy/hypershift-azure-destroy-chain.metadata.json [hunk 3]
```diff
@@ -6,7 +6,8 @@
 			"enxebre",
 			"sjenning",
 			"imain",
-			"davidvossel"
+			"davidvossel",
```
[the remark this answers is not in the document]
Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/azure/destroy/hypershift-azure-destroy-chain.metadata.json:9]: I added myself here: `ci-operator/step-registry/hypershift/OWNERS`, Then i run command: `make registry-metadata`, Which will help me automatically sync up to all these files.

**Pull request merged**